### ISSUE TITLE
BO - Loading order statuses is too slow (PS 1.7.6.3) - caused by PrestashopLogger - a proposal on acceleration

### ISSUE THREAD

**Issue by Eva-F** [PrestaShop/PrestaShop#20856]
#### Is your feature request related to a problem?
I needed to check OrderStatuses in BO
I had to wait a long time to the list loading (almost **2minutes**!!!!!!!)

See picture bellow

**the loading time: 1.9 minutes**

![ps20](https://user-images.githubusercontent.com/3242659/92220802-cb386100-ee9c-11ea-948f-da80b7558fc9.png)

after bellow suggested changes

**the loading time: 9.4 secs**
![ps22](https://user-images.githubusercontent.com/3242659/92228738-a5b15480-eea8-11ea-835a-dd5374e814b8.png)

it's not a hit, but it's an acceptable time

#### Describe the solution you'd like
I have looked for who caused such enomerous time in such relative simple list (the table ps_order_state has 38 records)

The most time was spent in the function thumbnail in classes/ImageManager.php - when the state of image is logged (the table ps_log)
I cannot judge the substantiation of all logs in the function thumbnail, but certainly exists the way to speed up them.

for the specification: 
the log table of my client contains approx. 224000 records - it is not imho a huge table ( but huge enough to the fact that contains no indexes) , but unfortunatelly the core of searching lies in searching of message, that is of type text. 
Adding record itself into ps_log isn't problem, the problem is in the function **isPresent()** in classes/PrestaShopLogger.php:

````
protected function isPresent()
    {
        if (!isset(self::$is_present[md5($this->message)])) {
            self::$is_present[$this->getHash()] = Db::getInstance()->getValue('SELECT COUNT(*)
				FROM `' . _DB_PREFIX_ . 'log`
				WHERE
					`message` = \'' . $this->message . '\'
					AND `severity` = \'' . $this->severity . '\'
					AND `error_code` = \'' . $this->error_code . '\'
					AND `object_type` = \'' . $this->object_type . '\'
					AND `object_id` = \'' . $this->object_id . '\'
				');
        }
        return self::$is_present[$this->getHash()];
    }
````
I ran sql of typical message  from the function **thumbnail** with profiles
![ps21](https://user-images.githubusercontent.com/3242659/92226754-849b3480-eea5-11ea-86ec-b21ba34919d1.png)

(if there is multiplied this time by the count of orderstates and count of logs in the function thumbnail) then the time near 2 minutes of loading corresponds.)

lets plaý with this "isPresent" sql:

1. add index
 ALTER TABLE `ps_log`
  ADD PRIMARY KEY (`id_log`), ADD KEY `log_options` (`severity`,`error_code`,`object_type`,`object_id`);
 ![ps23](https://user-images.githubusercontent.com/3242659/92229560-f83f4080-eea9-11ea-93db-a018ea50efd6.png)
 _it seems not a very useful - even worse_

2. specify object_type
 ![ps24](https://user-images.githubusercontent.com/3242659/92229843-61bf4f00-eeaa-11ea-9895-422b7879eafe.png)
_what a difference_ !!

**the conclusion:**
the OrderStatuses loading does faster:
1. adding index to columns  (`severity`,`error_code`,`object_type`,`object_id`) into ps_log
2. specify object_type in  in the function thumbnail in classes/ImageManager.php (i would reccomended to specify type for all PrestashopLogger->addLog calling in all Admin classes/controllers)
f,e,
````
Logger::addLog('not file_exists :' . $image,1,0,'Image',0);
````
3. Log only exceptions and unsuccesful cases in the similiar functions as the function "thumbnail" in classes/ImageManager.php

#### Additional context

small remark:
the best (imho) expression, that object_id is unspecified is to set its value to 0.
Therefore is neccessary to accept "0" as valid object_Id 
It can be realized by the change "if" condition  in the function addLog (line 133) in classes/PrestaShopLogger.php
````
//        if (!empty($objectType) && !empty($objectId)) {  <-- origin
        if (!empty($objectType) && !is_null($objectId)) {
            $log->object_type = pSQL($objectType);
            $log->object_id = (int) $objectId;
        }

````

**Comment by SimonGrn** [comment]
Hello,
thanks for your research and your proposal.
I see you used PrestaShop 1.7.6.3. In the next version to come (1.7.7.0) we entirely remade the whole Orders page, so it would be interesting to see if you could try with the same data (beta 2 was launched [a few days ago](https://build.prestashop.com/news/prestashop-1-7-7-0-beta2-release/)) to at least verify all these problems are still here.
What do you think ?

**Comment by Eva-F** [comment]
I met this problem (isPresent() function in PrestashopLogger) when I tried to load Orderstates.
but not only this controller(AdminStatuses)  uses thumbnail function and not only thumbnail() function uses PrestashopLogger->addLog().
I have tried to describe the big delaying caused by PrestashopLogger::isPresent() function
It is not issue related to Order, it concerns PrestashopLogger

**Comment by SimonGrn** [comment]
I understand that, but we make modifications on the code everyday and version 1.7.6.3 is more than one year old and it's really possible we fixed this in the last version, so it could be beneficial to check in 1.7.7.0. That's why it would be a very good idea just to see if it's not already been fixed.
If the problem is still here, we won't make any new patch version for 1.7.6.x anyway and it'll be fixed in 1.7.8 (develop).

**Comment by khouloudbelguith** [comment]
Hi @Eva-F,

Since we had no news from you for more than 20 days, I'll close this ticket. Feel free to open another one if you can give specific details.

Thanks!
